File: lang/mixed/euler-solutions/1/1.py
# (c) Simon Liedtke <liedtke.simon@googlemail.com> under the ISC license.
# See COPYING for more details.


def main():
    return sum(num for num in range(1000) if any(not num % x for x in [3, 5]))


if __name__ == '__main__':
    print(main())

File: lang/challenges/euler-solutions/1/1.clj
; (c) Florian Mayer <flormayer@aim.com> under the ISC license.
; See COPYING for more details.

(defn divides? [n x] (zero? (rem x n)))
(defn any-divides? [ns x] (some #(divides? % x) ns))
(defn get-divided [ns lst] (filter (partial any-divides? ns) lst))

(def sum (partial apply +))
(println (sum (get-divided [3 5] (range 1000))))
